### DISCUSSION TITLE
unzipping pymupdf on snowflake

### DISCUSSION THREAD

**Discussion by ksahoocbre** [pymupdf/PyMuPDF#3120] [Q&A]
### Description of the bug

I am trying to use pymupdf on snowflake using the zipped file from the main branch since pymupdf is not natively supported on Snowflake. 
The issue is I am getting a relative import error when importing the `fitz` module. The `from . import *` from src/fitz.py is causing the error. I've already raised an issue on the snowflake forum (please see below link) with the code snippet 
[issue](https://community.snowflake.com/s/question/0D7VI000000hDur0AE/detail?fromEmail=1&s1oid=00Di0000000hZh2&s1nid=0DB3100000001Fq&s1uid=005Do0000030CAw&s1ext=0&emkind=chatterCommentNotification&emtm=1706387230203)

As you may see, the solution provided by the snowflake representative is to fix this import.
I looked into it and seems I may need to edit the `src/fitz.py` file. 
Any help in this regard is highly appreciated

Best
Kamlesh

### How to reproduce the bug

Please see the posted code snippet [here](https://community.snowflake.com/s/question/0D7VI000000hDur0AE/detail?fromEmail=1&s1oid=00Di0000000hZh2&s1nid=0DB3100000001Fq&s1uid=005Do0000030CAw&s1ext=0&emkind=chatterCommentNotification&emtm=1706387230203)

### PyMuPDF version

1.23.19

### Operating system

Other

### Python version

3.11

**Comment by JorjMcKie** [comment]
This post seems best for the Discussions tab for further communication.

**Comment by julian-smith-artifex-com** [comment]
It looks like you're trying to use PyMuPDF directly from the unzipped source without building and installing it.

This will not work, and the error message you're seeing is expected. PyMuPDF has a lot of C and C++ code that needs to be compiled and linked and then everything needs to be installed properly; you cannot just use it directly.

I don't know how one builds and installs packages in Snowflake, so cannot advise you what to do - you'll need to ask the Snowflake people i think.

On a plain Python system, one would do `pip install pymupdf`; depending on how Snowflake works, something like this might work, but it's possible that you'll need to define some sort of Snowflake PyMuPDF package that uses PyMuPDF's `setup.py` script directly.
